### PR TITLE
Added function names to all constructors.

### PR DESCRIPTION
**Pull request by arodic**
This PR adds constructor names to previously anonymous constructors in THREE namespace. There should be no breaking changes in this pull request.

For example, previously `myPerspCamera.constructor.name` would return `""`. 

With this change it will return, `"PerspectiveCamera"`.

### CODE REVIEW

**Comment by arodic** [comment]
Removed the type property and updated internal logic to use `constructor.name` instead. Editor and examples appear to work as expected. Are there any tests I should run against this PR?

Note that this is a breaking change since it removes existing property from many objects. If merged, change log should include a warning.

Existing JSON scenes should load without a problem since `constructor.name` still maps to `type` property in JSON.

File: src/loaders/ObjectLoader.js [hunk 1]
```diff
@@ -90,6 +90,8 @@ THREE.ObjectLoader.prototype = {
 				var geometry;
 				var data = json[ i ];
 
+				console.log(data.type );
```
Comment (mrdoob) [inline src/loaders/ObjectLoader.js]: Left over.

Comment (mrdoob) [inline src/loaders/ObjectLoader.js]: Actually, I'll clean it on my end.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**